test/_agent.js

```diff
@@ -8,13 +8,35 @@ var Filters = require('object-filter-sequence')
 var uncaughtExceptionListeners = process._events.uncaughtException
 var agent
 
-module.exports = setup
+const noopFn = () => {}
+class NoopLogger {
+  warn () { return noopFn }
+  debug () { return noopFn }
+  info () { return noopFn }
+  fatal () { return noopFn }
+  error () { return noopFn }
+}
+
+class AgentWithNoopLogger extends Agent {
+  constructor () {
+    super()
+    this.defaultLogger = new NoopLogger()
+    Object.defineProperty(this, 'logger', {
```

Comment:
the reason for doing this is because, even if we do `agent._config({ logger: new NoopLogger()} ), It later gets changed in individual tests with original logger which does not  seem to be an ideal behaviour.